main.go

```diff
@@ -58,81 +60,135 @@ type config struct {
+	flags.StringVar(&c.ServingOptions.ServerCert.CertKey.KeyFile, "tls-private-key-file", "", "File containing the default x509 private key matching --tls-cert-file.")
+	flags.StringVar(&c.ServingOptions.MinTLSVersion, "tls-min-version", "VersionTLS12", "Minimum TLS version supported. Value must match version names from https://golang.org/pkg/crypto/tls/#pkg-constants.")
+	flags.StringSliceVar(&c.ServingOptions.CipherSuites, "tls-cipher-suites", nil, "Comma-separated list of cipher suites for the server. Values are from tls package constants (https://golang.org/pkg/crypto/tls/#pkg-constants). If omitted, the default Go cipher suites will be used")
 
 	// Auth flags
-	flagset.StringVar(&cfg.auth.Authentication.X509.ClientCAFile, "client-ca-file", "", "If set, any request presenting a client certificate signed by one of the authorities in the client-ca-file is authenticated with an identity corresponding to the CommonName of the client certificate.")
-	flagset.BoolVar(&cfg.auth.Authentication.Header.Enabled, "auth-header-fields-enabled", false, "When set to true, kube-rbac-proxy adds auth-related fields to the headers of http requests sent to the upstream")
-	flagset.StringVar(&cfg.auth.Authentication.Header.UserFieldName, "auth-header-user-field-name", "x-remote-user", "The name of the field inside a http(2) request header to tell the upstream server about the user's name")
-	flagset.StringVar(&cfg.auth.Authentication.Header.GroupsFieldName, "auth-header-groups-field-name", "x-remote-groups", "The name of the field inside a http(2) request header to tell the upstream server about the user's groups")
-	flagset.StringVar(&cfg.auth.Authentication.Header.GroupSeparator, "auth-header-groups-field-separator", "|", "The separator string used for concatenating multiple group names in a groups header field's value")
-	flagset.StringSliceVar(&cfg.auth.Authentication.Token.Audiences, "auth-token-audiences", []string{}, "Comma-separated list of token audiences to accept. By default a token does not have to have any specific audience. It is recommended to set a specific audience.")
+	flags.StringVar(&c.AuthenticationOptions.ClientCert.ClientCA, "client-ca-file", "", "If set, any request presenting a client certificate signed by one of the authorities in the client-ca-file is authenticated with an identity corresponding to the CommonName of the client certificate.")
+	flags.BoolVar(&c.AuthenticationOptions.SkipInClusterLookup, "authentication-skip-lookup", true, ""+
+		"If false, the authentication-kubeconfig will be used to lookup missing authentication "+
+		"configuration from the cluster.")
+	flags.BoolVar(&c.auth.Authentication.Header.Enabled, "auth-header-fields-enabled", false, "When set to true, kube-rbac-proxy adds auth-related fields to the headers of http requests sent to the upstream")
```

Comment:
this wiring looks incorrect.  Is this used to determine how to handle requests arriving at the kube-rbac-proxy or used to determine how to send requests to the thing kube-rbac-proxy is sitting in front of?

Reply:
> this wiring looks incorrect. Is this used to determine how to handle requests arriving at the kube-rbac-proxy or used to determine how to send requests to the thing kube-rbac-proxy is sitting in front of?

Oh, it only diffed due to name changes. nm

Reply:
It is used to determine how to send requests from kube-rbac-proxy to upstream. Therefore this is independent of the apiserver request header authentication flags. Thus I left this as it is.